### PR TITLE
msftidy and misc. fixups for Quest BoF module

### PR DESCRIPTION
**Pull request by busterb**
Just some fixes that got missed because msftidy in master has more checks.
Just need to verify it still works as expected.

#8223

### CODE REVIEW

File: modules/exploits/linux/misc/quest_pmmasterd_bof.rb [hunk 1]
```diff
@@ -3,8 +3,6 @@
 # Current source: https://github.com/rapid7/metasploit-framework
 ##
 
-require 'msf/core'
-
 class MetasploitModule < Msf::Exploit::Remote
   Rank = NormalRanking
```
Comment (busterb) [inline modules/exploits/linux/misc/quest_pmmasterd_bof.rb]: should this be Normal?

File: modules/exploits/linux/misc/quest_pmmasterd_bof.rb [hunk 2]
```diff
@@ -33,11 +33,10 @@ def initialize(info = {})
       'References'     =>
         [
           ['CVE', '2017-6553'],
-          ['URL' , 'https://0xdeadface.wordpress.com/2017/04/07/multiple-vulnerabilities-in-quest-privilege-manager-6-0-0-xx-cve-2017-6553-cve-2017-6554/']
+          ['URL', 'https://0xdeadface.wordpress.com/2017/04/07/multiple-vulnerabilities-in-quest-privilege-manager-6-0-0-xx-cve-2017-6553-cve-2017-6554/']
         ],
       'Payload'        =>
         {
-          'BadChars'    => "",
```
Comment (busterb) [inline modules/exploits/linux/misc/quest_pmmasterd_bof.rb]: I don't think we need to explicitly specify there are no badchars

File: modules/exploits/linux/misc/quest_pmmasterd_bof.rb [hunk 3]
```diff
@@ -50,41 +49,49 @@ def initialize(info = {})
...
-
-    register_options([ Opt::RPORT(12345) ], self.class)
-    register_options( [ Opt::CPORT(rand(1024))], self.class )
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(12345),
+        Opt::CPORT(rand(1024))
```
Comment (egypt) [inline modules/exploits/linux/misc/quest_pmmasterd_bof.rb]: Is this strictly necessary for the bug? It will make the module require root privs.

Comment (busterb) [inline modules/exploits/linux/misc/quest_pmmasterd_bof.rb]: not necessarily (see how docker images do it) - but I also don't know if it is necessary for the bug either

Comment (busterb) [inline modules/exploits/linux/misc/quest_pmmasterd_bof.rb]: https://github.com/rapid7/metasploit-framework/pull/8223#discussion_r115076104
Yes, it is needed

**Comment by wvu-r7** [comment]
Works for me. Thanks for taking care of it.

File: modules/exploits/linux/misc/quest_pmmasterd_bof.rb (hunk 1 above)
Comment (wvu-r7) [inline modules/exploits/linux/misc/quest_pmmasterd_bof.rb]: `NormalRanking` sounds right.

**Comment by wvu-r7** [comment]
# Release Notes

This cleans up the `exploit/linux/misc/quest_pmmasterd_bof` module to be `msftidy`-compliant. Additionally, the automatic targeting logic was adjusted to skip modules that set `DefaultTarget`.

**Pull request merged**